src/commands/convert.js

```diff
@@ -1,176 +1,247 @@
+    return Promise.resolve();
+  }
+
+  /**
+   * @param {Object} cfg
+   * @return {Promise}
+   * @private
+   */
+  _toJson(cfg) {
+    const componentConfigPath = join(ConvertCommand._buildComponentPath(cfg), config.defaultFileName);
+    const rawConfig = ConfigLoader.readConfig(componentConfigPath);
+    
+    if (!this._checkIfFilesIsJson(cfg)) {      
+      if (!rawConfig.component.hasOwnProperty('template')) {
+        return ConvertCommand._revertComponent(cfg).then(() => { 
+          return ConvertCommand._saveComponentJson(cfg).then(() => {
+            this.logSuccess(cfg.name, 'JSON'); 
+          }); 
+        });      
+      }
+      return ConvertCommand._saveComponentJson(cfg).then(() => {
+        this.logSuccess(cfg.name, 'JSON'); 
+      }); 
+    }
+
+    this.logSkip(cfg.name, 'JSON');
+    return Promise.resolve();
+  }
+
+  /**
+   * @param {Object} config
+   * @param {String} format
+   * @return {Promise}
+   * @private
+   */
+  _convertComponent(config, format) {
+    let promise;
+
+    switch (format) {
+      case 'yaml':
+      case 'yml':
+        promise = this._toYml(config);
+        break;
+
+      case 'hcl':
+        promise = this._toHcl(config);
+        break;
+
+      case 'json':
+        promise = this._toJson(config);
+        break;
+    }
+
+    return promise;
+  }
+
+  /**
+   * @param {String} name
+   * @param {String} format
+   */
+  logSuccess(name, format) {
+    this.logger.info(`Component '${name}' was successfully converted in ${format} format.`);
+  }
+
+  /**
+   * @param {String} name
+   * @param {String} format
+   */
+  logSkip(name, format) {
+    this.logger.warn(`Component '${name}' is already in ${format} format.`);
+  }
+
+  /**
+   * @param {Object} config
+   * @return {String}
+   * @private
+   */
+  static _buildComponentPath(config) {
+    return join(config.project.root, config.root);
+  }
+
+  /**
+   * @param format
+   * @private
+   */
+  static _validateFormat(format) {
+    if (!ConvertCommand._supportedFormats.includes(format)) {
+      throw new Error(`'${format}' format is not supported for configuration conversion.`);
+    }
+  }
+
+  /**
+   * @param {Object} config
+   * @return {Promise}
+   * @private
+   */
+  static _saveComponent(config) {
+    const configPath = ConvertCommand._buildComponentPath(config);
+
+    const tmpPath = homePath(jitPath);
+    const arch = Downloader.getOsArch();
+    const componentBinPath = join(binPath, arch);
+
+    return exec(`${join(componentBinPath, 'component')} -thub ${tmpPath} ${configPath} ${config.name}`);
+  }
+
+  /**
+   * @param {Object} config
+   * @return {Boolean}
+   * @private
+   */
+  _checkIfFilesIsJson(config) {
+    const configPath = ConvertCommand._buildComponentPath(config);
+    const mainFilePath = `${configPath}${sep}main.tf`;
+    
+    if (!fse.existsSync(mainFilePath)) {
+      return false;
+    }
+
+    let rawdata = fse.readFileSync(mainFilePath);  

```

Comment:
If you do
```suggestion
    let rawdata = fse.readFileSync(mainFilePath, 'utf8');
```
you don't need to use `toString` method to cast result to string.